pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -650,8 +671,11 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 	util.AppendLbEndpointMetadata(meta, ep.Metadata)
 
 	waypoint := ""
-	address, port := e.Address, int(e.EndpointPort)
-	tunnel := supportTunnel(b, e)
+	addresses, port := e.Addresses, int(e.EndpointPort)
+	tunnel, supportTunnelAddr := supportTunnel(b, e)
```

Comment:
I am very confused how this works. Can we explain in code comments the expected behavior of dual stack + waypoint?

Reply:
Hi @howardjohn, I compared with the original code and logic to do such changes, and I also check the waypoint which still is in single stack deployment, so I change the code in `supportTunnel` to return the first IP of the waypoint if `e` is waypoint.

Reply:
BTW, I also add more comments in code for this logic and the behavior of dual stack +waypoint.